### ISSUE TITLE
more import issues...

### ISSUE THREAD

**Issue by kbwestfall** [PYPIT/PYPIT#313]
I went a bit hog-wild and moved all imports to the top of all files for all of the pypit code in my new import_cleanup branch.  I'm testing it and it's faulting, as one might expect, but this particular failure mode is very weird and not a circular import issue as far as I can tell.  Here's part of the pytest output:

```
>       arcparm = ararc.setup_param(slf, 0, 1, fitsdict)

test_ararc.py:37:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../ararc.py:327: in setup_param
    wvmnx=arcparam['wvmnx'], modify_parse_dict=modify_dict)
../ararclines.py:77: in load_arcline_list
    tbls.append(tbl[['Ion','wave','RelInt']])
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/table/table.py:1228: in __getitem__
    out = self.__class__([self[x] for x in item],
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/table/table.py:1228: in <listcomp>
    out = self.__class__([self[x] for x in item],
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/table/table.py:1222: in __getitem__
    return self.columns[item]

...

>           return OrderedDict.__getitem__(self, item)
E           KeyError: 'RelInt'

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/astropy/table/table.py:106: KeyError
```

It's faulting because `'RelInt'` was not a valid key in the `astropy.table.Table` object.  I dug a bit and found that if I change the keyword from `'RelInt'` to `'Rel.'` (see [here](https://github.com/PYPIT/PYPIT/blob/165e81a1860b6ebcdf9b5a8d0ad7b1c88b8206c7/pypit/ararclines.py#L77) and [here](https://github.com/PYPIT/PYPIT/blob/165e81a1860b6ebcdf9b5a8d0ad7b1c88b8206c7/pypit/ararclines.py#L141)) then it passes.

Somehow moving around the imports has changed what's in the tables passed to the functions being tests.  I could have screwed something up in the quick editing I've done, but I wanted to see if you had any first impressions...

**Comment by kbwestfall** [comment]
This may actually be an issue with my version of arclines.  I ran the same test with my kbwdev branch (currently identical to master) and I get the same error.  My version of arclines is up-to-date with arclines/master, and I can't checkout the circular_imports branch.  Maybe the latter is a permissions issue?  I can just wait and try this again when circular_imports has been merged with arclines/master.

**Comment by rcooke-ast** [comment]
You now have access to arclines, in case you want to check!

**Comment by profxj** [comment]
yes, this is related to a change in arclines.
get on its 'circular_imports' branch which is pending a PR.

**Comment by kbwestfall** [comment]
Great. Now the tests pass.  I was actually working off my arclines fork which is why I couldn't check out the circular_imports branch.  Now I'm on the right repo!

I'm not going to close this yet because I may have a linetools import issue, but I need to some time to narrow it down.  I'll post again once I know for sure.

**Comment by kbwestfall** [comment]
I've gotten all the circular imports worked out again.  I'll submit another pull request soon.  I'm still getting a linetools error though, e.g.:

```
ImportError while importing test module '/Users/westfall/Work/packages/PYPIT/pypit/tests/test_scripts.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test_scripts.py:19: in <module>
    from pypit.scripts import coadd_1dspec
../scripts/coadd_1dspec.py:22: in <module>
    from pypit import arcoadd
../arcoadd.py:18: in <module>
    from linetools.spectra.utils import collate
E   ImportError: No module named utils
```

This only happens in python 2.  I get the same error if I just try to `from linetools.spectra.utils import collate` from a python session.  Any ideas?

**Comment by profxj** [comment]
that's quite an odd one.
there is no way there is a hook 
from linetools back to pypit..

Will see if I can figure it out.

**Comment by kbwestfall** [comment]
I don't think it's a circular import between pypit and linetools.  It's internal to linetools, or my installation of it.  This is what I get with python2:

```
$ python
Python 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 12:01:12)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from linetools.spectra.utils import collate
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named utils
>>> from linetools.spectra import utils
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name utils
>>> from linetools import spectra
>>> spectra
<module 'linetools.spectra' from '/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/linetools/spectra/__init__.pyc'>
```

**Comment by profxj** [comment]
Right you are.

**Comment by kbwestfall** [comment]
Hmm...  Maybe it's a problem with my installation.  I've had some trouble with my python2 installation getting confused by the MacOS system python.  I thought I had fixed it but maybe not.  I'll setup the pull request and then we can work through you testing it on your machine and Travis.

**Comment by profxj** [comment]
sounds good.  Aside from the fact that Travis is now misbehaving..

**Comment by kbwestfall** [comment]
Yeah, I don't know where these Seg Faults are coming from ...

**Comment by rcooke-ast** [comment]
I've encountered a few Seg Faults in the cython code (but I think that's now removed entirely). I'm pretty sure that matplotlib produces seg faults sometimes, but I have never figured out why. I sometimes encounter it in another package that I've developed, when matplotlib trys to plot something (but only sometimes).

I'm not sure this comment helps at all...

**Comment by rcooke-ast** [comment]
By the way, my linetools is working just fine too. I get no errors on that import. I'm using python 2 and:

import linetools
linetools.__version__
u'0.3.dev1877'

**Comment by kbwestfall** [comment]
Weird.  Here's what I get:

```
$ python
Python 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 12:01:12)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import linetools
>>> linetools.version
<module 'linetools.version' from '/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/linetools/version.pyc'>
>>>
```

**Comment by profxj** [comment]
Am 99% sure it is Travis' fault.
I reran a test on Master that was passing
and it now fails.

**Comment by profxj** [comment]
As for linetools, there is a circular import of sorts
there that I could work on fixing but it would be a 
lot easier if it were breaking for me!  ;)

**Comment by kbwestfall** [comment]
Is there a way that you could clone and install it in a fresh environment?  Like with pyenv or something?  Otherwise, I could play around with it next week.

**Comment by profxj** [comment]
Well addressed by PR #314
